### ISSUE TITLE
Autotools, etc. need updating for Apple M1 (arm64)

### ISSUE THREAD

**Issue by davidfetter** [pgbouncer/pgbouncer#644]
Expected:
```shell
/autogen.sh
Using aclocal: 1.16
```
Got:
```shell
./autogen.sh
Using aclocal: 1.16
configure.ac:15: warning: The macro `AC_PROG_CC_STDC' is obsolete.
configure.ac:15: You should run autoupdate.
./lib/autoconf/c.m4:1666: AC_PROG_CC_STDC is expanded from...
lib/m4/usual.m4:69: AC_USUAL_PROGRAM_CHECK is expanded from...
configure.ac:15: the top level
configure.ac:15: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:15: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
lib/m4sugar/m4sh.m4:692: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
lib/m4/usual.m4:69: AC_USUAL_PROGRAM_CHECK is expanded from...
configure.ac:15: the top level
configure.ac:191: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:191: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
lib/m4sugar/m4sh.m4:692: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
lib/m4/usual.m4:329: AC_USUAL_GETADDRINFO_A is expanded from...
configure.ac:191: the top level
configure.ac:191: warning: $as_echo is obsolete; use AS_ECHO(["message"]) instead
lib/m4sugar/m4sh.m4:692: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
lib/m4/ax_pthread.m4:89: AX_PTHREAD is expanded from...
lib/m4/usual.m4:329: AC_USUAL_GETADDRINFO_A is expanded from...
configure.ac:191: the top level
```

**Comment by petere** [comment]
This looks like it is from Autoconf 2.71, which is relatively new. Note that these are just obsolescence warnings; everything still works.

**Comment by yahavb** [comment]
```
[pgbouncer]$./configure ...
configure: WARNING: you should use --build, --host, --target
checking build system type... Invalid configuration `...': machine `...' not recognized
configure: error: /bin/sh ./config.sub ... failed
[pgbouncer]$./configure ..
configure: WARNING: you should use --build, --host, --target
checking build system type... Invalid configuration `..': machine `..' not recognized
configure: error: /bin/sh ./config.sub .. failed
[pgbouncer]$./configure 
checking build system type... arm-apple-darwin20.6.0
checking host system type... arm-apple-darwin20.6.0
checking target host type... unix
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking for a BSD-compatible install... /usr/bin/install -c
checking for grep that handles long lines and -e... /usr/bin/grep
checking for a race-free mkdir -p... ./install-sh -c -d
checking how to run the C preprocessor... gcc -E
checking whether compiler supports __func__... yes
checking whether linker supports --as-needed... no
checking whether compiler supports dependency generation... yes
checking for working warning switches... done
checking whether ln -s works... yes
checking for egrep... /usr/bin/grep -E
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking for a sed that does not truncate output... /usr/bin/sed
checking for strip... strip
checking for ranlib... ranlib
checking for ar... ar
checking for pkg-config... /opt/homebrew/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for pandoc... pandoc
checking for python3... python3
checking for strip... (cached) strip
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for inttypes.h... (cached) yes
checking for stdbool.h... yes
checking for unistd.h... (cached) yes
checking for sys/time.h... yes
checking for sys/socket.h... yes
checking for poll.h... yes
checking for sys/un.h... yes
checking for arpa/inet.h... yes
checking for netinet/in.h... yes
checking for netinet/tcp.h... yes
checking for sys/param.h... yes
checking for sys/uio.h... yes
checking for pwd.h... yes
checking for grp.h... yes
checking for sys/wait.h... yes
checking for sys/mman.h... yes
checking for syslog.h... yes
checking for netdb.h... yes
checking for dlfcn.h... yes
checking for err.h... yes
checking for pthread.h... yes
checking for endian.h... no
checking for sys/endian.h... no
checking for byteswap.h... no
checking for malloc.h... no
checking for regex.h... yes
checking for getopt.h... yes
checking for fnmatch.h... yes
checking for langinfo.h... yes
checking for xlocale.h... yes
checking for linux/random.h... no
checking for ucred.h... no
checking for sys/ucred.h... yes
checking for sys/resource.h... yes
checking for sys/wait.h... (cached) yes
checking for inline... inline
checking for C/C++ restrict keyword... __restrict__
checking whether byte ordering is bigendian... no
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... no
checking for pid_t... yes
checking for uid_t in sys/types.h... yes
checking for size_t... yes
checking for uint8_t... yes
checking for uint32_t... yes
checking for uint64_t... yes
checking for gcc options needed to detect all undeclared functions... none needed
checking whether strerror_r is declared... yes
checking for strlcpy... yes
checking for strlcat... yes
checking for strnlen... yes
checking for strsep... yes
checking for getpeereid... yes
checking for sigaction... yes
checking for sigqueue... no
checking for memmem... yes
checking for memrchr... no
checking for mempcpy... no
checking for inet_ntop... yes
checking for inet_pton... yes
checking for poll... yes
checking for getline... yes
checking for regcomp... yes
checking for err... yes
checking for errx... yes
checking for warn... yes
checking for warnx... yes
checking for getprogname... yes
checking for setprogname... yes
checking for posix_memalign... yes
checking for memalign... no
checking for valloc... yes
checking for explicit_bzero... no
checking for memset_s... yes
checking for reallocarray... no
checking for getopt... yes
checking for getopt_long... yes
checking for getopt_long_only... yes
checking for fls... yes
checking for flsl... yes
checking for flsll... yes
checking for ffs... yes
checking for ffsl... yes
checking for ffsll... yes
checking for fnmatch... yes
checking for mbsnrtowcs... yes
checking for nl_langinfo... yes
checking for strtod_l... yes
checking for strtonum... yes
checking for asprintf... yes
checking for vasprintf... yes
checking for timegm... yes
checking for localtime_r... yes
checking for gettimeofday... yes
checking for recvmsg... yes
checking for sendmsg... yes
checking for usleep... yes
checking for getrusage... yes
checking for syslog... yes
checking for mmap... yes
checking for getpeerucred... no
checking for arc4random_buf... yes
checking for getentropy... yes
checking for getrandom... no
checking for library containing WSAGetLastError... no
checking whether strerror_r returns char *... no
checking for integer enc/dec functions... not found
checking for library containing clock_gettime... none required
checking for library containing getsockname... none required
checking for library containing gethostbyname... none required
checking for library containing hstrerror... none required
checking for lstat... yes
checking for libevent... yes
checking whether to build with systemd support... no
checking whether to use c-ares for DNS lookups... auto
checking for libcares >= 1.6.0... yes
checking for ares_parse_soa_reply... yes
checking for OpenSSL... configure: error: not found
```
but I validated OpenSSL is installed 
```
[pgbouncer]$brew install OpenSSL
Warning: openssl@3 3.0.0_1 is already installed and up-to-date.
To reinstall 3.0.0_1, run:
  brew reinstall openssl@3
```

Not sure if I can blame arm64 for that or anything else is needed. 

I try to build pgbouncer for graviton (arm) instances.

**Comment by petere** [comment]
@yahavb Your problem is probably an outdated `config.guess` and `config.sub`.

**Comment by petere** [comment]
The originally reported problem has been addressed by libusual/libusual@4e4e580e3e5418f6f5b23fd09ae59b24fdb7cb27. It still has the warning about `AC_PROG_CC_STDC` being obsolete, but I don't plan to change that right now, because that would break/remove support for Autoconf 2.69, which is what we've been using so far, and that would be gratuitous for the sake of cleanliness at this point, I think.
